File: Fleetcode_Backend/FleetCode-main/models/squad.py
from mongoengine import Document, StringField, IntField, DateTimeField
from datetime import datetime

class Squad(Document):
    squad_name = StringField(required=True, unique=True)
    score = IntField(default=0)
    streak = IntField(default=0)
    spotter_tokens = IntField(default=1)
    created_at = DateTimeField(default=datetime.utcnow)

File: fleetcode-ui/backend/models/Squad.js
const mongoose = require('mongoose');

const SquadSchema = new mongoose.Schema({
  name: { type: String, required: true, unique: true }, // e.g., "Tee-Toddlers"
  score: { type: Number, default: 0 }, // Total XP for the Leaderboard
  streak: { type: Number, default: 0 },
  spotterTokens: { type: Number, default: 1 }, // Used in SpotterModal.jsx
  members: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }]
}, { timestamps: true });

module.exports = mongoose.model('Squad', SquadSchema);